src/coreComponents/constitutive/thermalConductivity/SinglePhaseThermalConductivity.cpp

```diff
@@ -0,0 +1,123 @@
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SinglePhaseThermalConductivity.cpp
+ */
+
+#include "SinglePhaseThermalConductivity.hpp"
+
+namespace geos
+{
+
+using namespace dataRepository;
+
+namespace constitutive
+{
+
+SinglePhaseThermalConductivity::SinglePhaseThermalConductivity( string const & name, Group * const parent ):
+  SinglePhaseThermalConductivityBase( name, parent )
+{
+  registerWrapper( viewKeyStruct::defaultThermalConductivityComponentsString(), &m_defaultThermalConductivityComponents ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setRestartFlags( RestartFlags::NO_WRITE ).
+    setDescription( "xx, yy, and zz diagonal components of the default thermal conductivity tensor [J/(s.m.K)]" );
+
+  registerWrapper( viewKeyStruct::thermalConductivityGradientComponentsString(), &m_thermalConductivityGradientComponents ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setApplyDefaultValue( {0.0, 0.0, 0.0} ).
+    setDescription( "xx, yy, and zz diagonal components of the thermal conductivity gradient tensor w.r.t. temperature [J/(s.m.K^2)]" );
+
+  registerWrapper( viewKeyStruct::referenceTemperatureString(), &m_referenceTemperature ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setApplyDefaultValue( 0.0 ).
+    setDescription( "The reference temperature at which the conductivity components are equal to the default values" );
+}
+
+std::unique_ptr< ConstitutiveBase >
+SinglePhaseThermalConductivity::deliverClone( string const & name,
+                                              Group * const parent ) const
+{
+  return SinglePhaseThermalConductivityBase::deliverClone( name, parent );
+}
+
+void SinglePhaseThermalConductivity::initializeRockFluidState( arrayView2d< real64 const > const & initialPorosity ) const
+{
+  for( localIndex ei = 0; ei < initialPorosity.size( 0 ); ++ei )
+  {
+    // NOTE: enforcing 1 quadrature point
+    for( localIndex q = 0; q < 1; ++q )
+    {
+      m_effectiveConductivity[ei][q][0] = m_defaultThermalConductivityComponents[0];
+      m_effectiveConductivity[ei][q][1] = m_defaultThermalConductivityComponents[1];
+      m_effectiveConductivity[ei][q][2] = m_defaultThermalConductivityComponents[2];
+
+      m_dEffectiveConductivity_dT[ei][q][0] = m_thermalConductivityGradientComponents[0];
+      m_dEffectiveConductivity_dT[ei][q][1] = m_thermalConductivityGradientComponents[1];
+      m_dEffectiveConductivity_dT[ei][q][2] = m_thermalConductivityGradientComponents[2];
+    }
+  }
+}
+
+void SinglePhaseThermalConductivity::updateFromTemperature( arrayView1d< real64 const > const & temperature ) const
+{
+  for( localIndex ei = 0; ei < temperature.size( 0 ); ++ei )
```

Comment:
```suggestion
  forAll<ParallelDevicePolicy<>>( temperature.size( 0 ), [=] GEOS_HOST_DEVICE ( localIndex const ei )
  {
  
  });
```

Reply:
and then you need to create views for all those member variables your are using inside the loop coz you cannot use member variables inside that raja loop.

For example:

````c++
auto effectiveConductivity = m_effectiveConductivity.toView();
````

Reply:
Done